### ISSUE TITLE
Confusing instructions for resolving conflicts

### ISSUE THREAD

**Issue by jrosdahl** [ingydotnet/git-subrepo#121]
Let's say I get a conflict when pulling a subrepo:

```
% git subrepo pull example
The rebase command failed:
----
First, rewinding head to replay your work on top of it...
Applying: Working on introducing the new login framework.
Using index info to reconstruct a base tree...
M	file1.txt
Falling back to patching base and 3-way merge...
Auto-merging file1.txt
Applying: git subrepo pull example
Using index info to reconstruct a base tree...
M	file2.txt
Falling back to patching base and 3-way merge...
Auto-merging file2.txt
CONFLICT (content): Merge conflict in file2.txt
Patch failed at 0002 git subrepo pull example
The copy of the patch that failed is found in:
   /some/directory/.git/rebase-apply/patch

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".
----
The 'git rebase' command failed during your pull.
You will need to finish the pull by hand. Follow these steps:

  # Resolve the rebase conflicts
  git checkout master
  git subrepo commit example

To abort the pull and reset back to where you started:

  git rebase --abort
  git checkout ORIG_HEAD
  git subrepo clean example

See 'git help subrepo' for more help.
```

I think that the instructions could be improved in both scenarios ("resolve conflicts" and "abort).

### 1. Aborting

The instructions for aborting don't work well for me:

```
% git rebase --abort
% git checkout ORIG_HEAD
Note: checking out 'ORIG_HEAD'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at deadbee... git subrepo pull example
% git subrepo clean example
git-subrepo: Must be on a branch to run this command.
```

I guess that the branch that I was on before doing `subrepo pull` should be checked out instead of `ORIG_HEAD`?

### 2. Resolving conflicts

If you aren't used to how `git rebase` works, or at least not what steps to take to resolve rebase conflicts, I think that git-subrepo's instructions will be quite confusing.

### 3. Suggestion

Here's an improvement idea:

```
The "git rebase" command failed during your pull.
You will need to finish the pull by hand. Follow these steps:

  1. Solve the conflicts (see "git status").
  2. "git add" the resolved files.
  3. git rebase --continue
  4. If there are more conflicts, start at step 1 again.
  5. git checkout master
  6. git subrepo commit example

Alternatively, you can abort the pull and reset back to where you started:

  1. git rebase --abort
  2. git checkout master
  3. git subrepo clean example
```

Thoughts?

**Comment by grimmySwe** [comment]
Looks good.

I guess that the `master`in the example above will be replaced with the actual branch name?

**Comment by ingydotnet** [comment]
Can someone PR this for me? I'm having a hard time digesting it all at the
moment. Better instructions on failure are always welcome.

**Comment by ingydotnet** [comment]
Looking over old issues #31 seems related. Thoughts?

**Comment by grimmySwe** [comment]
I think that Joels suggestion is a good first step, if you encounter
conflicts you are most likely forced to solve. Reset as in #31 might be
useful but I think that is lower priority.

fredag 22 januari 2016 skrev Ingy döt Net <notifications@github.com>:

**Comment by jrosdahl** [comment]
> Can someone PR this for me? I'm having a hard time digesting it all at the moment. Better instructions on failure are always welcome.

Sure, done.

**Comment by ingydotnet** [comment]
Fixed in #126
